ext/pgsql/pgsql.c

```diff
@@ -4250,7 +4268,14 @@ PHP_PGSQL_API zend_result php_pgsql_meta_data(PGconn *pg_link, const zend_string
 		name = PQgetvalue(pg_result,i,0);
 		add_assoc_zval(meta, name, &elem);
 	}
+
+	add_assoc_bool_ex(meta, pg_cache_field, sizeof(pg_cache_field) - 1, false);
+
+	tmp = (zval *)safe_emalloc(1, sizeof(zval), 0);
+	ZVAL_COPY(tmp, meta);
+	zend_hash_add(zmeta, (zend_string *)table_name, tmp);
```

Comment:
```suggestion
	zend_hash_add(zmeta, table_name, tmp);
```

Reply:
in fact locally I opted for the zend_hash_str* api to avoid this.